### PR TITLE
add inner to numpy 

### PR DESCRIPTION
**Pull request by ShiftLove**

### CODE REVIEW

**Comment by ShiftLove** [comment]
@simonetgordon Hey there. Have you seen this?

**Review by simonetgordon — CHANGES_REQUESTED** [review]
At first glance the code looks ok, however it looks like you've added new files instead of placing the functions in the appropriate pre-existing files? Please have a check over the contributor guide for instructions on this https://lets-unify.ai/ivy/deep_dive/16_ivy_frontends.html. I imagine it will go with other linear algebra functions.

**Comment by ShiftLove** [comment]
@simonetgordon I have reviewed and placed the functions in their appropriate file locations.
The function can be found here: https://github.com/ShiftLove/ivy/blob/master/ivy_tests/test_ivy/test_frontends/test_numpy/test_linear_algebra/test_inner

While its corresponding test function can be found here: ivy/functional/frontends/numpy/linear_algebra/inner

**Review by simonetgordon — COMMENTED** [review]
Thanks for moving them! I just needed to add a label to the PR so that the numpy frontend tests will run. Once this has finished, you can check the 'Details' of the check to make sure test_inner is passing.

**Comment by simonetgordon** [comment]
Also, don't forget to comment 'Close #Issue_number’, as explained [here](https://lets-unify.ai/ivy/contributing/1_the_basics.html#todo-list-issues) for this PR 😊

**Comment by ShiftLove** [comment]
Thank you for your smooth direction @simonetgordon . 

Close #4243

**Review by simonetgordon — CHANGES_REQUESTED** [review]
these functions should be in `ivy/functional/frontends/numpy/linear_algebra/matrix_and_vector_products.py` i.e. shadowing outer()

**Comment by simonetgordon** [comment]
I just checked the folders that already exist and these need to be moved to the pre-existing file indicated in above

**Comment by ShiftLove** [comment]
> I just checked the folders that already exist and these need to be moved to the pre-existing file indicated in above

@simonetgordon what you mean is that I should move the functions from the new folders that I created and paste them in to the pre-existing folder, just below the outer function, right?

**Review by simonetgordon — CHANGES_REQUESTED** [review]
Remove `ivy/functional/frontends/numpy/linear_algebra/inner` file and move test_inner to test_matrix_and_vector_products.py

**Comment by ShiftLove** [comment]
> Remove `ivy/functional/frontends/numpy/linear_algebra/inner` file and move test_inner to test_matrix_and_vector_products.py

@simonetgordon Again, done.

File: ivy/functional/frontends/numpy/linear_algebra/matrix_and_vector_products.py [hunk 1]
```diff
@@ -4,3 +4,13 @@
 
 def outer(a, b, out=None):
     return ivy.outer(a, b, out=out)
+
+
+#inner
```
Comment (simonetgordon) [inline ivy/functional/frontends/numpy/linear_algebra/matrix_and_vector_products.py:9]: no need to import ivy again down here. Also remove the comments like #global and # inner. Just have 2 lines between outer and inner functions.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_linear_algebra/test_matrix_and_vector_products.py [hunk 2]
```diff
@@ -75,3 +75,84 @@ def test_numpy_outer(
         b=np.array(y, dtype=input_dtype2),
         out=None,
     )
+    
+    
+# global
```
Comment (simonetgordon) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_linear_algebra/test_matrix_and_vector_products.py:80]: remove comment and just have two blank lines here

File: ivy/functional/frontends/numpy/linear_algebra/matrix_and_vector_products.py [hunk 3]
```diff
@@ -4,3 +4,13 @@
...
+
+# global
+import ivy
+
+
+def inner(a, b, out=None):
+    return ivy.inner(a, b, out=out)
```
Comment (simonetgordon) [inline ivy/functional/frontends/numpy/linear_algebra/matrix_and_vector_products.py:16]: This is calling a backend implementation which didn't exist- I have just added it. Please merge with master to get this update.

**Review by simonetgordon — CHANGES_REQUESTED** [review]

**Comment by ShiftLove** [comment]
@simonetgordon Thanks for your patience. Individual Changes resolved.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_linear_algebra/test_matrix_and_vector_products.py [hunk 4]
```diff
@@ -75,3 +75,82 @@ def test_numpy_outer(
         b=np.array(y, dtype=input_dtype2),
         out=None,
     )
+    
+
+import numpy as np
```
Comment (simonetgordon) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_linear_algebra/test_matrix_and_vector_products.py:80]: Same for all of these imports. All imports go at the top of the file and there should be no duplication.

**Review by simonetgordon — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_linear_algebra/test_matrix_and_vector_products.py (hunk 4 above)
Comment (ShiftLove) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_linear_algebra/test_matrix_and_vector_products.py]: @simonetgordon  I have corrected that.

**Comment by ShiftLove** [comment]
Hi @simonetgordon ,
Are we good now?

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_linear_algebra/test_matrix_and_vector_products.py [hunk 5]
```diff
@@ -75,3 +75,74 @@ def test_numpy_outer(
...
+
+    data1 = (input_dtype, x)
+
+    shape = (
+        draw(st.integers(min_value=1, max_value=5)),
+        draw(st.integers(min_value=1, max_value=5)),
+    )
+    data2 = draw(
+        helpers.dtype_and_values(
+            available_dtypes=ivy_np.valid_numeric_dtypes,
```
Comment (simonetgordon) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_linear_algebra/test_matrix_and_vector_products.py:107]: `ivy_np` is not recognised i.e. hasn't been imported properly.

**Review by simonetgordon — CHANGES_REQUESTED** [review]

**Comment by simonetgordon** [comment]
Hi there, check the details of the lint test for more import issues

**Comment by ShiftLove** [comment]
> Hi there, check the details of the lint test for more import issues

Goodmorning @simonetgordon my lint test has been checked and completed successfully.

**Comment by simonetgordon** [comment]
Nice! Once the frontend test has finished, check `Details` -> `Run Frontend Tests` and search for `inner` to see if it's passing this test.

**Review by simonetgordon — APPROVED** [review]
Looks good!

**Comment by simonetgordon** [comment]
Closes #4243

**Pull request merged**